test/unit-tests/expect-tests/to-throw-async.spec.ts

```diff
@@ -0,0 +1,222 @@
+import { AsyncTest, Expect, TestCase } from "../../../core/alsatian-core";
+import { ErrorMatchError } from "../../../core/errors/error-match-error";
+
+export class ToThrowAsyncTests {
+   // Asynchronous throw
+   private async asyncThrowFunction(delayMs: number): Promise<void> {
+      return new Promise<void>((_, reject) => {
+         setTimeout(reject(new Error("Timeout then reject")), delayMs);
+      });
+   }
+
+   // Asynchronous non-throw
+   private async asyncNonThrowFunction(delayMs: number): Promise<void> {
+      return new Promise<void>((resolve) => {
+         setTimeout(resolve(), delayMs);
+      });
+   }
+
+   @TestCase(0)
+   @TestCase(100)
+   @AsyncTest()
+   public async asyncFunctionThrowsErrorPasses(delayMs: number) {
+      // Exect to and error NOT be thrown
+      await Expect(async() => {
+         // Expect an error to be thrown
+         await Expect(async () => this.asyncThrowFunction(delayMs)).toThrowAsync();
+      }).not.toThrowAsync();
+   }
+
+   @TestCase(0)
+   @TestCase(100)
+   @AsyncTest()
```

Comment:
Could we add test descriptions please :) trying to add more of these as more readable than the function name